core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/handlers/HandlerService.java

```diff
@@ -79,7 +79,7 @@ private static void check(Set<Class> handlerTargets, Class target) {
     checkArgument(handlerTargets.contains(target), "No Handlers for " + target.getSimpleName() + " defined");
   }
 
-  private <T extends ManagedType> Handler<T> getHandler(T managedType) {
+  private <T extends Modeled> Handler<T> getHandler(T managedType) {
```

Comment:
Not keen on `Modeled`. I assume the 'data' equivalent is still `ManagedType`?
How about `Entity` for the (public facing) service layer (nice, simple, clear) and `DataEntity` for the repository layer?

Reply:
We can't change ManagedType, and I don't feel that Entity really conveys anything. Everything is an Entity.

Can we leave this for now?

Reply:
Once this interface is extracted and being used externally we'll never come back and change it so I'd prefer we do it now.

Why can't we change `ManagedType`? `Entity` conveys more than `ManagedType` which is a bit cryptic. Says nothing about. At least Entity says what it is. And the model and data interfaces should at least have some equivalent in their naming - `ManagedType` and `Modeled` have none.

Reply:
ManagedType is an interface in the persistence package, we don't get to change that.

Reply:
Happy to hear suggestions of what would convey the intent better than Modeled.
To me a 'Modeled' Object is one that exists in the core Model. It has a state and a specification. What should we call it instead?

Reply:
Changed to Entity and DataEntity